curse modpack importing and fix caching

NikkyAI opened the pull request.

NikkyAI committed:
command: import

NikkyAI committed:
process files WIP

NikkyAI committed:
generate config based on imported zip and init git repo

blacklist modID "examplemod"
added `git init` to the init command 
updating works with changed files now.. as long as you do not want to switch to a different version / branch

NikkyAI committed:
made CurseMeta IDisposable

removed try cach block lef over from testing

NikkyAI committed:
hacked in a cache fix

NikkyAI commented:
caching was fixed in a prrof of concept way
feel free to convert that into actualy IDisposables or whatever fits @copygirl

copygirl commented on src/Alpacka.Lib/Curse/ResourceHandlerCurse.cs:
```diff
@@ -52,60 +52,62 @@ public class ResourceHandlerCurse : IResourceHandler
+                            Name    = depAddon.Name,
+                            Handler = Name,
+                            Source  = dep.AddonId.ToString(),
+                            Version = Release.Latest.ToString(), // avoid crashes from listing files
```
`avoid crashes from listing files` - does this still apply?

NikkyAI replied:
it does no longer apply.. actually this refers to a behaviour that was removed when i moved the curse api stuff into alpacka-meta, it used to either "just" get the recommended file
or get all files and look for the best one to pick

now it always gets info for all files per mod since its the same effort, it includes all deleted files we know about.. and its is safe

NikkyAI committed:
simplify class structure of EntryResource

allows for extension with interfaces

NikkyAI committed:
use recommended as default version until the proper defaults can be applied

NikkyAI committed:
updated documentation

copygirl committed:
Implement WriteYaml for Defaults and Includes

Also add Serializer to ModpackConfig

NikkyAI committed:
reverse my attempt to improve things

NikkyAI committed:
serializing generated config to yaml is still broken

NikkyAI committed:
Fix the yamlWriter again

i keep on saying its unecessary complexity

NikkyAI committed:
use IValueSerializer inside WriteYaml

NikkyAI committed:
fix caching.. again

this time trying a Singleton approach, looks good so far

copygirl commented on src/Alpacka.Lib/Pack/Config/ModpackConfig.cs:
```diff
@@ -42,5 +40,18 @@ public static ModpackConfig LoadYAML(string path)
+        public void SaveYAML(string path)
+        {
+            if (!Directory.Exists(path))
+                Directory.CreateDirectory(path);
```
It's safe to call `CreateDirectory` if the directory already exists so you can remove the check.

NikkyAI committed:
apply prettification

NikkyAI committed:
added tons of stuff

usage documented in commands.md
extended functionality of multimc handler with icon support
added fallback for more forgeversions
